apps/rush-lib/src/utilities/TarExecutable.ts

```diff
@@ -28,37 +45,64 @@ export class TarExecutable {
    * @returns
    * The "tar" exit code
    */
-  public async tryUntarAsync(archivePath: string, outputFolderPath: string): Promise<number> {
-    const childProcess: ChildProcess = Executable.spawn(this._tarExecutablePath, ['-x', '-f', archivePath], {
-      currentWorkingDirectory: outputFolderPath
-    });
-    const [tarExitCode] = await events.once(childProcess, 'exit');
-    return tarExitCode;
+  public async tryUntarAsync(options: IUntarOptions): Promise<number> {
+    return await this._spawnTarWithLoggingAsync(
+      ['-x', '-f', options.archivePath],
+      options.outputFolderPath,
+      options.logFilePath
+    );
   }
 
   /**
    * @returns
    * The "tar" exit code
    */
-  public async tryCreateArchiveFromProjectPathsAsync(
-    archivePath: string,
-    paths: string[],
-    project: RushConfigurationProject
-  ): Promise<number> {
+  public async tryCreateArchiveFromProjectPathsAsync(options: ICreateArchiveOptions): Promise<number> {
+    const { project, archivePath, paths, logFilePath } = options;
     const pathsListFilePath: string = `${project.projectRushTempFolder}/tarPaths_${Date.now()}`;
     await FileSystem.writeFileAsync(pathsListFilePath, paths.join('\n'));
 
     const projectFolderPath: string = project.projectFolder;
-    const childProcess: ChildProcess = Executable.spawn(
-      this._tarExecutablePath,
+    const tarExitCode: number = await this._spawnTarWithLoggingAsync(
       ['-c', '-f', archivePath, '-z', '-C', projectFolderPath, '--files-from', pathsListFilePath],
-      {
-        currentWorkingDirectory: projectFolderPath
-      }
+      projectFolderPath,
+      logFilePath
     );
-    const [tarExitCode] = await events.once(childProcess, 'exit');
     await FileSystem.deleteFileAsync(pathsListFilePath);
 
     return tarExitCode;
   }
+
+  private async _spawnTarWithLoggingAsync(
+    args: string[],
+    currentWorkingDirectory: string,
+    logFilePath: string
+  ): Promise<number> {
+    await FileSystem.ensureFolderAsync(path.dirname(logFilePath));
+    const fileWriter: FileWriter = FileWriter.open(logFilePath);
+    fileWriter.write(
+      [
+        `Invoking "${this._tarExecutablePath} ${args.join(' ')}"`,
+        '',
+        '======= BEGIN PROCESS OUTPUT =======',
+        ''
+      ].join('\n')
+    );
+
+    const childProcess: ChildProcess = Executable.spawn(this._tarExecutablePath, args, {
+      currentWorkingDirectory: currentWorkingDirectory
+    });
+
+    childProcess.stdout.on('data', (chunk) => fileWriter.write(`[stdout] ${chunk}`));
+    childProcess.stderr.on('data', (chunk) => fileWriter.write(`[stderr] ${chunk}`));
```

Comment:
I don't know. This is really only for investigative purposes, so it doesn't really matter that much since someone doing the investigation can peel the chunks back apart manually.